src/main.cpp

```diff
@@ -2239,16 +2239,23 @@ CAmount GetBlockSubsidy(int nHeight, const Consensus::Params& consensusParams)
     }
 }
 
+static std::atomic<bool> IBDLatchToFalse{false};
+// testing-only, allow initial block down state to be set or reset
+bool TestSetIBD(bool ibd) {
+    bool ret = !IBDLatchToFalse;
+    IBDLatchToFalse.store(!ibd, std::memory_order_relaxed);
+    return ret;
+}
+
 bool IsInitialBlockDownload(const Consensus::Params& params)
 {
     // Once this function has returned false, it must remain false.
-    static std::atomic<bool> latchToFalse{false};
     // Optimization: pre-test latch before taking the lock.
-    if (latchToFalse.load(std::memory_order_relaxed))
+    if (IBDLatchToFalse.load(std::memory_order_relaxed))
```

Comment:
Why not just access `IBDLatchToFalse` within the `cs_main` lock, and take that lock when modifying it?

Reply:
Oh, I see, the double-checked locking optimization was already there. Hmmph. (Bitcoin systemically makes concurrency design decisions that I disagree with, but okay, let's do the minimal change here.)